### ISSUE TITLE
Cannot create JWK from a Hash

### ISSUE THREAD

**Issue by akotranza** [jwt/ruby-jwt#532]
latest version of this library
Rails 6.1.x
Ruby 2.7.5

Question/Bug converting a hash to a JWK suitable for `JWT.decode`?

(And no, after several hours of lambdas absolutely not working I am 100% not interested in using `jwks`, I want to pass a single dang key with a specific `kid` that I have grabbed from the jwks response from the openid server)

`JWT::JWK.new(myHash)` throws `Cannot create JWK from a Hash`

said `myHash` looks something like
`{:kid=>"somestring", :kty=>"EC", :alg=>"ES256", :use=>"sig", :crv=>"P-256", :x=>"anotherstring", :y=>"yetanotherstring"}` so I'm not sure why exactly it's being rejected.

Expected:

```
key_as_hash = handwaveymagic() # returns {:kid=>"somestring", :kty=>"EC", :alg=>"ES256", :use=>"sig", :crv=>"P-256", :x=>"anotherstring", :y=>"yetanotherstring"}
key = JWT::JWK.new key_as_hash
decoded_token = JWT.decode(token, key, true, { iss: @issuer, verify_iss: true, algorithms: [alg] })
# hooray
```

Actual:
```
JWT::JWKError (Cannot create JWK from a Hash):
```

**Comment by anakinj** [comment]
Hi, thanks for the question.

Unfortunately the latest improvements for the JWKs have not been release yet, suggest checking out the documentation for the exact version you are using to avoid confusion. For example for v2.5.0: https://github.com/jwt/ruby-jwt/blob/v2.5.0/README.md

Using the latest released (v2.5.0) version you have at least two options.

1. Import the JWK and provide the keypair as the `key`
```
jwk = JWT::JWK.import(key_as_hash) # in the next version you will be able to do  JWT::JWK.new(key_as_hash)

decoded_token = JWT.decode(token, jwk.keypair, true, { iss: @issuer, verify_iss: true, algorithms: [alg] })
```

2. Provide the key set as a jwk set to the decode in the `jwks` parameter. (This approach requires the token to have a matching kid header)
```
decoded_token = JWT.decode(token, nil, true, { iss: @issuer, verify_iss: true, algorithms: [alg], jwks: {keys: [key_as_hash]}})
```

**Comment by akotranza** [comment]
Thank you for the detailed response! I ended up using a gem `jwk-loader` which seems to have overcome the issue I had figuring out exactly what `jwks` param is expecting to be returned when a proc/function is supplied. But I will likely be referring to your reply when writing tests 😄
